tests/test_ratelimiter/test_api.py

```diff
@@ -45,7 +45,7 @@ async def test__ratelimiter__limit_exceeded() -> None:
     async def calc() -> float:
         return 42
 
-    with pytest.raises(RateLimitExceeded):
+    with pytest.raises(EmptyBucket):
```

Comment:
Could we please keep the RateLimitExceeded exception? It's more tied to the ratelimiter "domain" while the bucket error feels like exposing detail of the underlying implementation to ratelimiter consumers.

Reply:
Yeah, so the tokenbucket class throws an empty bucket exception, and the token bucket rate limiter class essentially pulls that logic from the tokenbucket class. Should we just make the  acquire function in tokenbucketlimiter catch this exception and throw the ratelimitexceeded exception?